infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -128,7 +129,9 @@
     private long maxConflictWildcard = Address.NOT_FOUND;
 
     private final long defaultCacheSize = Long.MAX_VALUE;
-    private final Cache<Integer, Long> conflictToGlobalTailCache;
+    private final long defaultPerStreamCacheSize = Long.MAX_VALUE;
+    private long perStreamCacheSize = defaultPerStreamCacheSize;
+    private final Cache<UUID, Cache<Integer, Long>> conflictToGlobalTailCache;
```

Comment:
I am also somewhat concerned but for a different, concrete reason. 

Evicting conflict information on entire-stream basis seems somewhat crude. Whenever the cache limit is hit, I would prefer to evict many ancient entries from multiple streams, rather than all entries from one stream.

This is the case where I prefer to keep the conflict map "flat", and not sorted per stream. Yes, it is twice the space, but the cache eviction policy is more sane.